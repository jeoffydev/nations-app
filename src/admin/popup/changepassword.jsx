import React, {Component} from 'react';   
 
 
import Joi from 'joi-browser'; 


class ChangePassword extends Component{
    constructor(props){
        super(props); 
         
    } 

     
     
    state = {
        
        data: { 
             
        },   
         
    }

    //Joi schema
    schema = {  
    }

   

    componentDidMount(){   
      
       console.log("this.props.userId", this.props.userId)
        
    }
 

    doSubmit = async () =>{ 
         
    }

   
    
    
    
    render(){ 
        
        

        return (

            <React.Fragment> 

            <p> 
                <button type="button" className="btn btn-sm btn-dark " data-toggle="modal" data-target={'#ChangePasswordModal' + this.props.userId}> Change Password </button> 
            </p>

            
                  
            </React.Fragment>
        )
    }
}

export default ChangePassword;
 