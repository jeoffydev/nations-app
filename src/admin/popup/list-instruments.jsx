import React, {Component} from 'react';   
import { InputCheckBox } from './../../form/input';
 

const ListInstruments = ( {instrumentsArray, myItems,  onChange } ) =>{  

    //console.log("myItems", myItems);
    //console.log("instrumentsArray", instrumentsArray);

        var insSel = [];
        for (let value of myItems) {  
            insSel.push(value.instrumentId);
        }  

        const instrumentList = instrumentsArray.map( (ins) =>    
            <li className="list-group-item" key={ins.id}>   
                 {insSel.indexOf(ins.id) !== -1   ?  <InputCheckBox  name='instrumentCheckBox' id={'check' + ins.id} type='checkbox'   label={ins.instrumentName} value={ins.id} check={true}  onChange={onChange}     />   :   <InputCheckBox  name='instrumentCheckBox' id={'check' + ins.id} type='checkbox'   label={ins.instrumentName} value={ins.id} check={false}  onChange={onChange}     />  }   
            </li>
        ) 

    
        return (
                    
            <React.Fragment> 

                <ul className="list-group">     
                     {instrumentList}
                </ul> 


            </React.Fragment> 


        ) 
     
} 


export default ListInstruments;