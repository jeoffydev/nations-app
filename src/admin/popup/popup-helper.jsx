import React, {Component} from 'react';   

export function EditButton  ( props ) { 
    const {  idname,  id, label   } = props; 
    return (
            
        <React.Fragment>
            <div  > 
                <button type="button" className="btn btn-sm btn-dark " data-testid="editbutton" data-toggle="modal" data-target={'#' + idname + 'Modal' +  id}> {label} </button> 
            </div>
        </React.Fragment>
      
    )  

} 

export function AddButton  ( props ) { 
    const {  idname,  id, label   } = props; 
    return (
            
        <React.Fragment>
            <p> 
                <button type="button" className="btn   btn-dark " data-toggle="modal" data-target={'#' + idname + 'Modal' +  id}> {label} </button> 
            </p>
        </React.Fragment>
      
    )  

} 
 


export function PopupHeader  ( props ) {
     
    const {  idname,  label,   } = props;  
    return (
            
        <React.Fragment>
            <div className="modal-header">
                <h5 className="modal-title" id={ idname + "ModalLabel"}> {label}</h5>
                <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                    <span aria-hidden="true">&times;</span>
                </button>
            </div>
        </React.Fragment>
      
    )  
} 


export default {
    PopupHeader,
    EditButton
}