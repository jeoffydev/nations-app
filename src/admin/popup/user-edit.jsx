import React, {Component} from 'react';   
import Form from '../../form/form'; 
import Joi from 'joi-browser'; 

//Import these 2 for api and http post
import { apiEndPoint } from '../../config/apiEndPoint'; 
import axiosApiInstance from '../../auth/httpService'; 

import {PopupHeader, EditButton} from './popup-helper';

class UserEdit extends Form{
    constructor(props){
        super(props);

        
         
    } 

     
     
    state = {
        dataUsers : {},
        data: { 
            fullName: '',
            id: '',
            membersInstrumentViewModels: [] 
        },   
        instruments: {},
        insSel: [],
        errors : {},
    }

    //Joi schema
    schema = { 
        fullName : Joi.string().required().min(4),
        id : Joi.string().required(),
        membersInstrumentViewModels: Joi.array().required()  
    }

   

    componentDidMount(){   
         //Get Instruments in DB
         axiosApiInstance.get(apiEndPoint('get-instrument') )
         .then(res => {
             console.log(res.data);  
             this.setState( { instruments : res.data });
        }) 
        
        //Selected Instruments for Users
        const {selectedInstrument} = this.props; 
        //Select Instruments
        var insSel = [];
        for (let value of selectedInstrument) {  
            insSel.push(value.instrumentId);
        }  
         
        //User edit 
        const { data  } = this.state;
        const {item  } = this.props;  

        //User Edit
        data.fullName = this.props.item.fullname;
        data.id = this.props.item.id;
        //data.membersInstrumentViewModels = insSel;
        this.setState({data});

       
        
    }
 

    doSubmit = async () =>{ 
        const {   data } = this.state;
        console.log(  data );
        
        const dataObj = {
            "fullName" : data.fullName,
            "id": data.id 
        }
       console.log(apiEndPoint('update-user-details'));
        axiosApiInstance.put(apiEndPoint('update-user-details'),  dataObj )
        .then(res => {
            console.log(res.data);  
            
       })  
    }
    
    
    render(){ 

        const {instruments,   data  } = this.state;
        const {item, name } = this.props;   
        
        //Display all Instruments
        var insArray = [];
        for (let value of Object.values(instruments)) { 
            var arrayPush = {'id': value.id, 'instrumentName': value.instrumentName, 'instrumentDescription': value.instrumentDescription }
            insArray.push(arrayPush);
        } 
        //Display all Instruments in list
        const instrumentList = insArray.map( (ins) =>    
            <li className="list-group-item" key={ins.id}>  
                <div className="form-check">
                    {data.membersInstrumentViewModels.indexOf(ins.id) !== -1   ? this.InputCheckBox('instrumentCheckBox', 'check' + ins.id, 'checkbox', ins.instrumentName, ins.id, true)   :  this.InputCheckBox('instrumentCheckBox', 'check' + ins.id, 'checkbox', ins.instrumentName, ins.id, false) } 
                   
                </div>
            </li>
        ) 

        return (

            <React.Fragment>

                <EditButton idname={name} id={item.id} />
                
                {/* Model here */} 
                <div className="modal fade" id={name+ 'Modal' + item.id} tabIndex="-1" role="dialog" aria-labelledby="popModalLabel" aria-hidden="true">
                    <div className="modal-dialog" role="document">
                        <div className="modal-content">
                            <PopupHeader idname="user" label={item.fullname} /> 
                            <div className="modal-body"> 

                                    <form className="form-signin" onSubmit={this.handleSubmit} noValidate> 
                                        {this.renderInputEdit('id', '', 'hidden', item.id )} 
                                        {this.renderInputEditReadOnly('email', 'Email Address', 'email', item.email )} 
                                        {this.renderInputEdit('fullName', 'Full Name', 'text', data.fullName  )}

                                        <h3>Skills &amp; Instruments</h3>
                                        <ul className="list-group">     
                                            {instrumentList}
                                        </ul> 
                                        {this.renderButton('Update')} 
                                    </form>  
                            </div>
                            <div className="modal-footer">
                                <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button> 
                            </div>
                        </div>
                    </div>
                </div>
                {/* Model here */} 
                  
            </React.Fragment>
        )
    }
}

export default UserEdit;
 