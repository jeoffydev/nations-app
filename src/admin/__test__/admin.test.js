 
  
import React, {Component} from 'react';
import ReactDOM from 'react-dom'; 
import TestRenderer from 'react-test-renderer';
  
import Users from '../users';
import Songs from '../songs';
import Form from '../../form/input';
import AddInstrument from '../add-instrument';

//npm install @testing-library/react --save
import { render, cleanup } from '@testing-library/react';

import {PopupHeader, EditButton, AddButton} from '../popup/popup-helper';  

//Use @testing-library/jest-dom then import @testing-library/jest-dom/extend-expect
import "@testing-library/jest-dom/extend-expect";

import renderer   from 'react-test-renderer'; 

afterEach(cleanup);

beforeEach(() => {
    jest.spyOn(console, 'error')
    global.console.error.mockImplementation(() => {})
})

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<EditButton  label="Edit" />, div )
})


 
 

 
it("matches snapshots", ()=>{
    const tree = renderer.create( <EditButton    /> ) .toJSON()  ;
    expect(tree).toMatchSnapshot();
})


 

  const testRenderer = TestRenderer.create(<PopupHeader idname="song" label="Test label"  />);
  const testInstance = testRenderer.root;
  
  expect(testInstance.findByType(PopupHeader).props.idname).toBe('song');  