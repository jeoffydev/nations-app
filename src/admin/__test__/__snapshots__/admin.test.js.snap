// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div>
  <button
    className="btn btn-sm btn-dark "
    data-target="#undefinedModalundefined"
    data-testid="editbutton"
    data-toggle="modal"
    type="button"
  >
     
     
  </button>
</div>
`;
